src/Symfony/Component/Console/Output/ConsoleOutput.php

```diff
@@ -100,14 +101,32 @@ public function setErrorOutput(OutputInterface $error)
      * Returns true if current environment supports writing console output to
      * STDOUT.
      *
-     * IBM iSeries (OS400) exhibits character-encoding issues when writing to
-     * STDOUT and doesn't properly convert ASCII to EBCDIC, resulting in garbage
-     * output.
-     *
      * @return bool
      */
     protected function hasStdoutSupport()
     {
-        return ('OS400' != php_uname('s'));
+        return false === $this->runningOS400();
+    }
+
+    /**
+     * Returns true if current environment supports writing console output to
+     * STDERR.
+     *
+     * @return bool
+     */
+    protected function hasStderrSupport()
```

Comment:
@iltar If you have suggestions for existing objects where this logic might make sense please let me know. Else I feel moving this logic given a single edge case is over-abstraction.

While ConsoleOutput constructor accepts a STDOUT stream, it internally creates its own STDERR output stream object. Whether that's right or wrong, my current approach is very mindful of BC and fixes support for the widely used IBM platform.

Perhaps we keep this in mind for BC breaks in 3.0.